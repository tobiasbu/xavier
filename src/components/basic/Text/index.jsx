import React from 'react';
import PropTypes from 'prop-types';

/**
 * Astro text
 * @param {any} props Text props
 */
const Text = (props) => {
  const { size, children, className } = props;
  let sizeName = 'medium';
  if (size <= 1) {
    sizeName = 'large';
  } else if (size === 3) {
    sizeName = 'small';
  } else if (size >= 4) {
    sizeName = 'very-small';
  }
  return (
    <p className={`a-text--secondary-${sizeName} ${className}`}>{children}</p>
  );
};

/**
 * Text default props
 */
Text.defaultProps = {
  size: 2,
  children: '',
  className: '',
};

/**
 * Text prop types
 */
Text.propTypes = {
  size: PropTypes.number,
  children: PropTypes.node,
  className: PropTypes.string,
};

export default Text;
