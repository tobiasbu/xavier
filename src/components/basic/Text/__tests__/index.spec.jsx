import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Text from '..';

describe('Text', () => {
  it('should render', () => {
    const component = shallow(<Text size={4}>Uma palabra por-favor</Text>);
    expect(toJSON(component)).toMatchSnapshot();
  });
});
