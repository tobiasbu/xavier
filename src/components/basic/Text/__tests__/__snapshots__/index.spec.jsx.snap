// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render 1`] = `
<p
  className="a-text--secondary-very-small "
>
  Uma palabra por-favor
</p>
`;
